producer/writer/message_writer.go

```diff
@@ -475,27 +472,34 @@ func newAckHelper(size int) *acks {
 	}
 }
 
-func (h *acks) add(meta metadata, m *message) {
-	h.Lock()
-	h.m[meta] = m
-	h.Unlock()
+func (a *acks) add(meta metadata, m *message) {
+	a.Lock()
+	a.m[meta] = m
+	a.Unlock()
 }
 
-func (h *acks) remove(meta metadata) {
-	h.Lock()
-	delete(h.m, meta)
-	h.Unlock()
+func (a *acks) remove(meta metadata) {
+	a.Lock()
+	delete(a.m, meta)
+	a.Unlock()
 }
 
-func (h *acks) ack(meta metadata) {
-	h.Lock()
-	m, ok := h.m[meta]
+func (a *acks) ack(meta metadata) {
+	a.Lock()
+	m, ok := a.m[meta]
 	if !ok {
-		h.Unlock()
+		a.Unlock()
 		// Acking a message that is already acked, which is ok.
 		return
 	}
-	delete(h.m, meta)
-	h.Unlock()
+	delete(a.m, meta)
+	a.Unlock()
 	m.Ack()
 }
+
+func (a *acks) size() int {
+	a.Lock()
```

Comment:
nit: is it worth using a `RWLock` here?

Reply:
Not really, this `size()` is being called very rarely and all other functions on the hot path requires a write lock so I don't think rlock here would help much